src/atopile/cli/common.py

```diff
@@ -126,9 +127,38 @@ def create_build_contexts(
     build: Iterable[str],
     target: Iterable[str],
     option: Iterable[str],
+    no_project: bool,
 ) -> list[atopile.config.BuildContext]:
     entry, entry_arg_file_path = get_entry_arg_file_path(entry)
-    project_config = get_project_config(entry_arg_file_path)
+
+    if no_project:
+        if not entry:
+            raise errors.UserBadParameterError(
+                "You must specify an entry to build with the --no-project option"
+            )
+        if not entry_arg_file_path.exists():
+            raise errors.UserBadParameterError(
+                f"The file you have specified does not exist: {entry_arg_file_path}"
+            )
+        if not entry_arg_file_path.is_file():
+            raise errors.UserBadParameterError(
+                "The path you're building with the --no-project"
+                f" option must be a file {entry_arg_file_path}"
+            )
+        if not address.get_entry_section(entry):
+            raise errors.UserBadParameterError(
+                "You must specify what to build within a file to build with the"
+                " --no-project option"
+            )
+
+        project_config = atopile.config.ProjectConfig(
+            location=Path.cwd(),
+            ato_version=f"^{version.get_installed_atopile_version()}",
+            paths=atopile.config.ProjectPaths(layout=Path.cwd() / "projectless"),
+            builds={"default": atopile.config.ProjectBuildConfig(targets=[])},
```

Comment:
This has to default to no targets because otherwise we see some weird behaviour, for example the module map trying to index all the files in `atopile/`